tiledb/sm/serialization/query.cc

```diff
@@ -320,7 +344,9 @@ Status subarray_partitioner_from_capnp(
 
   // Overall mem budget
   RETURN_NOT_OK(partitioner->set_memory_budget(
-      reader.getMemoryBudget(), reader.getMemoryBudgetVar(), 0));
+      reader.getMemoryBudget(),
+      reader.getMemoryBudgetVar(),
+      reader.getMemoryBudgetValidity()));
```

Comment:
This probably needs to be wrapped in a `reader.hasMemoryBudgetValidity` so we can set a default `MemoryBudgetValidity` if it doesn't exist in the cap'n proto message.

Reply:
Likewise, there isn't a `hasMemoryBudgetValidity` routine. It defaults to zero, so the behavior here will be unchanged when unspecified. Leaving as-is.